cadc-vos/src/main/java/ca/nrc/cadc/vos/LinkNode.java

```diff
@@ -83,53 +84,39 @@ public class LinkNode extends Node {
     
     // A URI that points to any type of resource.
     private URI target;
-    
-    // TODO: 'structured' will be moved from Node to DataNode.
-    //       Removed this attribute after Node has been updated.
-    // Indicates if the VOSpace understands the data format.
-    private boolean structured;
 
     /**
      * Link node constructor.
-     * @param uri
-     * @param target
+     * @param name The name of the node.
+     * @param target The URI to the node resource.
      */
-    public LinkNode(VOSURI uri, URI target) {
-        super(uri);
-        this.structured = false;
-        this.target = target;
+    public LinkNode(String name, URI target) {
+        this(name, target, new TreeSet<>());
     }
 
     /**
      * LinkNode constructor.
-     * @param uri
-     * @param properties
-     * @param target
+     * @param name The name of the node.
+     * @param target The URI to the node resource.
+     * @param properties The node's properties.
      */
-    public LinkNode(VOSURI uri, List<NodeProperty> properties, URI target) {
-        super(uri, properties);
-        this.structured = false;
+    public LinkNode(String name, URI target, Set<NodeProperty> properties) {
+        super(name, properties);
+        NodeUtil.assertNotNull(LinkNode.class, "target", "target");
         this.target = target;
     }
 
-    /**
-     * @return true if the VOSpace understands the format of the data.
-     */
-    public boolean isStructured() {
-        return structured;
-    }
-
     /**
      * @return Target URI of this LinkNode instance.
      */    
     public URI getTarget() {
         return target;
     }
-    
+
     /**
      * Sets the target URI of this LinkNode instance.
-     * 
-     * @param target target URI.
+     *
+     * @param target The URI to the node resource.
```

Comment:
setTarget needs to enforce the not-null constraint

Reply:
Added a null check.